pkg/scheduler/algorithm/predicates/metadata.go

```diff
@@ -51,16 +51,7 @@ type topologyPair struct {
 	value string
 }
 
-type podSet map[*v1.Pod]struct{}
-
-type topologyPairSet map[topologyPair]struct{}
-
-// topologyPairsMaps keeps topologyPairToAntiAffinityPods and antiAffinityPodToTopologyPairs in sync
-// as they are the inverse of each others.
-type topologyPairsMaps struct {
-	topologyPairToPods map[topologyPair]podSet
-	podToTopologyPairs map[string]topologyPairSet
-}
+type topologyToMatchedTermCount map[topologyPair]int64
```

Comment:
Nvm, if possible, I believe it can be a followup along with comment:

https://github.com/kubernetes/kubernetes/blob/55f81314cc4f8ed08aab64d31d4f842587f2f17e/pkg/scheduler/algorithm/predicates/metadata.go#L443-L444

Reply:
Yeah, I thought about that, I did a mutex contention profile, and this can potentially bring ~10% improvement.

Reply:
+1 for isolating in a separate PR. There might be other forms of locking that we could consider too. Or using a channel.

Reply:
I'm not so sure that's thread safe. But let's leave the discussion for another PR :)

Reply:
I guess we will need to use an RLock:

```
mutex.RLock 
ptr := topologyToMatchedTermCount[pair]
mutex.RUnlock
if ptr  == nil {
  mutex.Lock()
  // we have to check again since by the time we get the lock, another thread might have already initialized the entry.
  if topologyToMatchedTermCount[pair]  == nil {
    topologyToMatchedTermCount[pair] = new(int64)
  } 
  ptr = topologyToMatchedTermCount[pair]
  mutex.Unlock ()
}

atomicAdd(ptr, value)
```